inmem/kv.go

```diff
@@ -23,22 +23,25 @@ const cursorBatchSize = 1000
 
 // KVStore is an in memory btree backed kv.Store.
 type KVStore struct {
-	mu      sync.RWMutex
+	mu      RWMutex
 	buckets map[string]*Bucket
 	ro      map[string]*bucket
 }
 
 // NewKVStore creates an instance of a KVStore.
 func NewKVStore() *KVStore {
 	return &KVStore{
+		mu:      NewLock(),
 		buckets: map[string]*Bucket{},
 		ro:      map[string]*bucket{},
 	}
 }
 
 // View opens up a transaction with a read lock.
 func (s *KVStore) View(ctx context.Context, fn func(kv.Tx) error) error {
-	s.mu.RLock()
+	if !s.mu.RTryLock(ctx) {
```

Comment:
This seems like a straightforward improvement.   Should we consider feature flagging this since it's on a service-critical path?